engine/core/Component.php

```diff
@@ -5,21 +5,21 @@ class Component
     public static array $_props = [];
     static function render(string $location, array $props = []): string
     {
-        self::$_props = $props;
         ob_start();
+        self::$_props[ob_get_level()] = $props;
```

Comment:
2-dimensional array allows for infinite component depth

I might build a more robust templating system in the future, taking inspiration from https://platesphp.com/